R/FilterStateExpr.R

```diff
@@ -0,0 +1,350 @@
+            if (isTRUE(private$is_disabled())) {
+              shinyWidgets::updateSwitchInput(inputId = "enable", value = FALSE)
+            }
+            if (isFALSE(private$is_disabled())) {
+              shinyWidgets::updateSwitchInput(inputId = "enable", value = TRUE)
+            }
+          })
+          out <- reactive(input$remove) # back to parent to remove self
+          out
+        }
+      )
+    },
+
+    #' @description
+    #' Shiny module UI.
+    #'
+    #' @param id (`character(1)`)\cr
+    #'  shiny element (module instance) id;
+    #'  the UI for this class contains simple message stating that it is not supported
+    #' @param parent_id (`character(1)`) id of the FilterStates card container
+    ui = function(id, parent_id = "cards") {
+      ns <- NS(id)
+
+      tags$div(
+        id = id,
+        class = "panel filter-card",
+        include_js_files("count-bar-labels.js"),
+        tags$div(
+          class = "filter-card-header",
+          tags$div(
+            class = "filter-card-title",
+            `data-toggle` = "collapse",
+            `data-bs-toggle` = "collapse",
+            href = paste0("#", ns("body")),
+            tags$span(tags$strong(private$id)),
+            tags$span(private$title, class = "filter-card-varlabel")
+          ),
+          tags$div(
+            class = "filter-card-controls",
+            shinyWidgets::prettySwitch(
+              ns("enable"),
+              label = "",
+              status = "success",
+              fill = TRUE,
+              value = !private$is_disabled(),
+              width = 30
+            ),
+            actionLink(
+              inputId = ns("remove"),
+              label = icon("circle-xmark", lib = "font-awesome"),
+              class = "filter-card-remove"
+            )
+          ),
+          tags$div(
+            class = "filter-card-summary",
+            `data-toggle` = "collapse",
+            `data-bs-toggle` = "collapse",
+            href = paste0("#", ns("body")),
+            private$ui_summary(ns("summary"))
+          )
+        )
+      )
+    },
+    #' @description
+    #' Destroy observers stored in `private$observers`.
+    #'
+    #' @return NULL invisibly
+    #'
+    destroy_observers = function() {
+      lapply(private$observers, function(x) x$destroy())
+      return(invisible(NULL))
+    }
+  ),
+  private = list(
+    dataname = character(0),
+    disabled = NULL,
+    expr = NULL,
+    extras = list(),
+    id = character(0),
+    observers = list(),
+    title = character(0),
+
+    # Disables `FilterState`
+    # `state` is moved to cache and set to `NULL`
+    # @return `NULL` invisibly
+    disable = function() {
+      private$disabled(TRUE)
+      invisible(NULL)
+    },
+
+    # Enables `FilterState`
+    # Cached `state` is reset again and cache is cleared.
+    # @return `NULL` invisibly
+    enable = function() {
+      private$disabled(FALSE)
+      invisible(NULL)
+    },
```

Comment:
Are these one-liner methods necessary if used only by `set_state`? I guess the same question can be asked for `FilterState`.